ledger/src/blockstore_processor.rs

```diff
@@ -294,6 +294,55 @@ fn execute_batches_internal(
     })
 }
 
+fn process_batches(
```

Comment:
so this fn is a drop-in replacement for the previous `execute_batches`.

In `process_entries` we still call this drop-in at the same times we called `execute_batches`, i.e. tick-boundaries, entry conflicts, end of loop.
Scheduling doesn't block on execution, and also cannot fail w/ an error - how does replay stage become aware of invalid block state without this fn returning it?

Reply:
with that in mind, i'm curious why the switch was introduced here instead of one layer up. Is there a benefit in collecting the until conflict/tick before scheduling?

Reply:
oh, thanks for well-deserved good questions! hope this in-source comments should answer to both of your questions: 28972802a7653b56d72a2da8a554d38bb3a44060

Reply:
I think so? I think the main reason is we want to share most of the same code since there are checks on non-self-conflicting entries. And since scheduling doesn't block on execution, there shouldn't be too much overhead from just collecting the multiple entries instead of scheduling them as they come in?

Reply:
yeah. that's correct.

Reply:
fyi, i a bit improved the in-source comment: 6e63d8742a7bf448adf8312c230b282e0037cfca